hybrid/per_func_comp_main.c

```diff
@@ -0,0 +1,257 @@
+/***
+ * This example implements a simple cheri type allocator to show
+ * how cheri otype values are determined.
+ * Note: cheriintrin.h is not used in this example.
+ ***/
+
+#include "../include/print.h"
+#include "include/utils.h"
+#include <sys/sysctl.h>
+#include <sys/types.h>
+
+#define SANDBOX_CODE_SZ 1000 // approx. later modify it.
+
+// NOTE ------
+/// cheriintrin.h is commented out temporarily in this example //
+typedef long cheri_otype_t;
+//#define cheri_is_sealed(x) __builtin_cheri_sealed_get(x)
+// from cheriintrin.h  ///
+
+// static void * __capability sandbox_1_sealcap;
+static void *__capability cheriobj_1_codecap;
```

Comment:
Could we add comments to say what these are for? I assume they are pairs of capabilities to define compartments, but not sure.

Reply:
As I said, the code is in progress, but currently showing making otypes and sealing code/data capabilities and printing the values.  
When the code is quite done, i expect to get rough idea for sandboxing using cheri.

Reply:
> As I said, the code is in progress

I thought this PR was complete in scope. Are there plans to add more features?